plugins/hatsu.ts

```diff
@@ -0,0 +1,101 @@
+import { merge } from "../core/utils/object.ts";
+import { read } from "../core/utils/read.ts";
+import { Page } from "../core/file.ts";
+
+import type Site from "../core/site.ts";
+
+export interface Options {
+  /**
+   * Hatsu instance URL.
+   *
+   * @default
+   * ```ts
+   * new URL('https://hatsu.local')
+   * ```
+   */
+  instance: URL;
+  /**
+   * Match routes.
+   *
+   * @default
+   * ```ts
+   * [/^\/posts\/(.+)$/]
+   * ```
+   */
+  matches?: (RegExp | string)[];
+  /**
+   * Whether to copy `/.well-known/` files.
+   * @default `undefined`
+   */
+  wellKnown?: false;
+}
+
+// Default options
+export const defaults: Options = {
+  instance: new URL("https://hatsu.local"),
+  matches: [/^\/posts\/(.+)$/],
+};
+
+export default (userOptions: Options) => {
+  const options = merge(defaults, userOptions);
+
+  return (site: Site) => {
+    if (options.matches) {
+      site.process([".html"], (pages) =>
```

Comment:
Instead of using regexp for urls, I think it's better to use a Search query. For example: `publish=true` to select all pages with the variable `publish` set to `true`.
You can get all pages with:
```
const pages = site.search.pages("publish=true")
```

Reply:
> Instead of using regexp for urls, I think it's better to use a Search query. For example: `publish=true` to select all pages with the variable `publish` set to `true`.

Indeed. is there a way to use a similar approach for middleware?

Reply:
Nope. Do you need it for the middleware?
I'm thinking of a solution similar to redirect plugin: it generates a _redirects.json file with all urls that you can import in the redirect middleware.

Reply:
> Do you need it for the middleware? 

Some software only supports redirects instead of HTML link alternates, and middleware redirects requests from them while leaving normal user requests intact.